### ISSUE TITLE
Exterior algebra: `⋆` is not a unary operator

### ISSUE THREAD

**Issue by rleegates** [JuliaLang/julia#17480]
I'm currently writing a library for [exterior algebra on manifolds](https://en.wikipedia.org/wiki/Exterior_algebra) and need to define the [Hodge star](https://en.wikipedia.org/wiki/Exterior_algebra#Hodge_duality) operator. In particular, the Hodge star maps a `k`-form on an `n`-dimensional manifold to an `n-k`-form. Is there any use-case which disagrees with `⋆` being a unary operator? If not and it's a quick fix, could fulfilling this request be part of the 0.5 release?

**Comment by simonbyrne** [comment]
You can call the function form of `*`, which accepts a single argument, by enclosing it in parentheses
```julia
julia> (*)(1)
1
```

**Comment by oxinabox** [comment]
We are talking about `\star` right?

What does work for `\star` is, like what @simonbyrne said for `*`, using it as a function.

```
julia> ⋆(x)=2*x
⋆ (generic function with 1 method)


julia> ⋆(4)
8
```

I believe [the current unary operators are defined here.](https://github.com/JuliaLang/julia/blob/master/src/julia-parser.scm#L65-L68)

**Comment by Keno** [comment]
I'm fine with letting `\star` be both unary and binary, but from a syntax perspective people may prefer to have it behave like `*`

**Comment by TotalVerb** [comment]
What about making unary `*` syntax for `unsafe_load`? (Joking. That would be terrible.)

On a more relevant note, how about making `\bigcup`, `\bigcap`, `\prod`, `\sum`, `\bigvee`, and `\bigwedge` unary? These symbols (⋃ ⋂ ∏ ∑ ⋁ ⋀) are fairly commonly used unary in mathematics.

**Comment by stevengj** [comment]
@TotalVerb, I think @JeffBezanson's perspective from way back was that `\cup` would be a binary operator and `\bigcup` (etc.) would just be an identifier name, so you can basically use it as a unary operator but you have to add parens.

**Comment by simonbyrne** [comment]
now we can pass comprehensions in functions, these should be reduction operators, i.e. `\bigcup (foo(i) for i in X)`

**Comment by TotalVerb** [comment]
I can get behind aliases like

```julia
const ∑ = sum
const ∏ = prod
⋀(x) = reduce(∧, x)  # all?
⋁(x) = reduce(∨, x)  # any?
⋃(x) = reduce(union, x)
⋂(x) = reduce(intersect, x)
```

**Comment by JeffreySarnoff** [comment]
wrt sets, Union( ∅, ∅ ) is Union( ∅ ) is (∅) which I am free to write this way ()  # math, not as a tuple
similarly Intersection, so how about
```julia
⋃(x) = reduce(union, x)
⋂(x) = reduce(intersect, x)

⋃() = Void()
⋂() = Void()

# and if it is useful to cover reducing over an empty collection, here are two
const emptyTuple = ()
const zeroElements = []

⋃(emptyTuple) = emptyTuple
⋂(emptyTuple) = emptyTuple
⋃(zeroElements) = zeroElements
⋂(zeroElements) = zeroElements
```

**Comment by c42f** [comment]
Fixed in #31604
